cmd/ttn-lw-cli/commands/use.go

```diff
@@ -0,0 +1,147 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"bytes"
+	"crypto/md5"
+	"crypto/tls"
+	"encoding/hex"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	errNoHost      = errors.DefineInvalidArgument("no_host", "no host set")
+	errFileExists  = errors.DefineAlreadyExists("file_exists", "`{file}` exists")
+	errFailWrite   = errors.DefinePermissionDenied("fail_write", "failed to write `{file}`")
+	configFileName = ".ttn-lw-cli.yml"
+
+	useCommand = &cobra.Command{
+		Use:               "use [host]",
+		Short:             "Use",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errNoHost
+			}
+			insecure, _ := cmd.Flags().GetBool("insecure")
+			ca, _ := cmd.Flags().GetBool("ca")
+			user, _ := cmd.Flags().GetBool("user")
+			overwrite, _ := cmd.Flags().GetBool("overwrite")
+
+			hash := func(in string) string {
+				h := md5.New()
+				io.Copy(h, strings.NewReader(in))
+				return hex.EncodeToString(h.Sum(nil))
+			}
+
+			destPath := func(base string, user bool, overwrite bool) (string, error) {
+				fileName := base
+				if user {
+					dir, err := os.UserConfigDir()
+					if err != nil {
+						return "", err
+					}
+					fileName = filepath.Join(dir, base)
+				}
+
+				_, err := os.Stat(fileName)
+				if os.IsNotExist(err) || overwrite {
+					return fileName, nil
+				}
+
+				logger.Warnf("%s already exists. Use --overwrite", fileName)
+				return "", errFileExists.WithAttributes("file", fileName)
+			}
+
+			caFileName := func(host string) string {
+				return fmt.Sprintf("ca.%s.pem", hash(host)[:6])
+			}
+
+			host := args[0]
+			conf := MakeDefaultConfig(host, insecure)
+			conf.CredentialsID = host
+
+			// Get CA certificate from server
+			if !insecure && ca {
+				logger.Infof("Will retrieve certificate from %s", conf.NetworkServerGRPCAddress)
+				caFile, err := destPath(caFileName(host), user, overwrite)
+				if err != nil {
+					return err
+				}
+
+				conn, err := tls.Dial("tcp", conf.NetworkServerGRPCAddress, &tls.Config{InsecureSkipVerify: true})
+				if err != nil {
+					return err
+				}
+				defer conn.Close()
+
+				var buf bytes.Buffer
```

Comment:
nit; to avoid doing `&buf` below in a loop, you could also just do
```suggestion
				buf := &bytes.Buffer{}
```